### DISCUSSION TITLE
Hibernate validator and reactive

### DISCUSSION THREAD

**Discussion by robp94** [quarkusio/quarkus#32275] [Q&A] [answered]
Is it somehow possible to have a custom validator with a blocking call, like to a web service, without blocking?
 
The interface ConstraintValidator does not have a Uni/CompletableFuture return type, there seems to be no reactive alternative.
If you use it anyway on reactive Methods with the `@Valid` annotation, it fails because it is blocking. The only ugly why I came up with is with using `@Blocking` annotation or calling it by myself and using for example the default worker pool of Mutiny.

Is there another option without blocking? Are there maybe plans to add a reactive version of the validator?

[a comment by an automated account is not shown]

**Comment by yrodiere** [comment, marked as answer]
You may be interested in this: https://github.com/quarkusio/quarkus/issues/28421#issuecomment-1271242939

Short answer: it's complicated, and Hibernate Validator/Bean Validation was not built for this.

As to future plans, we'll consider merging pull requests as always, as long as they take into account the constraints mentioned in the issue.

**Reply by robp94** [reply]
Ok unfortunate, but thx for the answer :)
